service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsControllerTest.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.C2DocumentBundle;
+import uk.gov.hmcts.reform.fpl.service.DateFormatterService;
+import uk.gov.hmcts.reform.fpl.service.UserDetailsService;
+
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.callbackRequest;
+import static uk.gov.hmcts.reform.fpl.utils.DocumentManagementStoreLoader.document;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(UploadC2DocumentsController.class)
+@OverrideAutoConfiguration(enabled = true)
+class UploadC2DocumentsControllerTest {
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+
+    @MockBean
+    private UserDetailsService userDetailsService;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private DateFormatterService dateFormatterService;
+
+    @Test
+    void shouldCreateC2DocumentBundleWithValidCaseData() throws Exception {
```

Comment:
Now on the contrary we are not testing the scenario when there are no documents. Are you happy with that? There is risk that with some refactoring NPE would be introduced. What do you think?

Reply:
As in we should be testing the scenario when c2 docs HAVE NOT been uploaded prior?

Reply:
Yes - as that's quite important case.